packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -583,14 +583,13 @@ class AndroidSdk {
     if (platform.isMacOS) {
       try {
         final String javaHomeOutput = processUtils.runSync(
-          <String>['/usr/libexec/java_home'],
+          <String>['/usr/libexec/java_home', '-v', '1.8'],
           throwOnError: true,
           hideStdout: true,
         ).stdout.trim();
         if (javaHomeOutput != null) {
-          final List<String> javaHomeOutputSplit = javaHomeOutput.split('\n');
-          if ((javaHomeOutputSplit != null) && (javaHomeOutputSplit.isNotEmpty)) {
-            final String javaHome = javaHomeOutputSplit[0].trim();
+          if ((javaHomeOutput != null) && (javaHomeOutput.isNotEmpty)) {
+            final String javaHome = javaHomeOutput.trim();
```

Comment:
```suggestion
            final String javaHome = javaHomeOutput.split('\n').last.trim();
```

Reply:
Running `java_home -v 999999`, outputs:
```
Unable to find any JVMs matching version "999999".
/Library/Java/JavaVirtualMachines/openjdk-12.0.1.jdk/Contents/Home
```
Therefore, it needs to extract the last line, which is the default JVM.

Reply:
Hello @blasten, you are suggesting this as a fix for when Java 8 is not installed?